llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -148,7 +148,6 @@ bool VPRecipeBase::mayHaveSideEffects() const {
   switch (getVPDefID()) {
   case VPDerivedIVSC:
   case VPPredInstPHISC:
-  case VPScalarCastSC:
```

Comment:
Do we need to handle `Instruction::isCast(getOpcode()` in `VPInstruction::opcodeMayReadOrWriteFromMemory()` so that scalar casts are still side effect free?

Reply:
Yes, updated, thanks!